packages/alfa-act/src/oracle.ts

```diff
@@ -4,7 +4,7 @@ import { Option } from "@siteimprove/alfa-option";
 import { Question } from "./question";
 import { Rule } from "./rule";
 
-export type Oracle<Q> = <I, T, A>(
-  rule: Rule<I, T, Q>,
-  question: { [K in keyof Q]: Question<K, Q[K], T, A> }[keyof Q]
-) => Future<Option<A>>;
+export type Oracle<Q> = <I, S, T>(
+  rule: Rule<I, S, Q>,
+  question: { [K in keyof Q]: Question<K, Q[K], S, T> }[keyof Q]
+) => Future<Option<T>>;
```

Comment:
Now they don't match `Rule` though 🙈 `I` is for `Input`, `T` is for `Target`, and `A` is for `Answer`.

Reply:
I guess, maybe Question should also have "targets" rather than "subjects" 😄 
(I spend a lot of time going between these type definitions trying to understand them and the mismatching type variable really bothered me… Another possibility could be to use longer type variable name (Target, Subject, …) instead of single letter ones, maybe full upper case (TARGET, SUBJECT, …) if we want to distinguish them visually)